artifactory/commands/transferfiles/utils.go

```diff
@@ -417,3 +418,120 @@ func stopTransferOnArtifactoryNodes(srcUpService *srcUserPluginService, runningN
 		}
 	}
 }
+
+// getMaxUniqueSnapshots gets the local repository's setting of max unique snapshots (Maven, Gradle, NuGet, Ivy and SBT)
+// or max unique tags (Docker).
+// On repositories of other package types or if an error is thrown, this function returns -1.
+func getMaxUniqueSnapshots(rtDetails *coreConfig.ServerDetails, repoSummary *serviceUtils.RepositorySummary) (maxUniqueSnapshots int, err error) {
+	maxUniqueSnapshots = -1
+	serviceManager, err := utils.CreateServiceManager(rtDetails, retries, retriesWaitMilliSecs, false)
+	if err != nil {
+		return
+	}
+	switch repoSummary.PackageType {
+	case maven:
```

Comment:
I think we can use the Factory Method design pattern in order to create concrete LocalRepositoryParam, and then we can reduce the number of duplicated lines, this also applies to 'updateMaxUniqueSnapshots'